pkg/detector/policy.go

```diff
@@ -175,6 +175,90 @@ func (d *ResourceDetector) getAndApplyClusterPolicy(object *unstructured.Unstruc
 	return d.ApplyClusterPolicy(object, objectKey, matchedClusterPropagationPolicy)
 }
 
+// removeSpecificResourceAndRBLabels remove the propagation policy labels of specific resource and corresponding binding.
+func (d *ResourceDetector) removeSpecificResourceAndRBLabels(resource *unstructured.Unstructured, removeLabels []string) error {
+	// 1. get rb of given resource
+	rbName := names.GenerateBindingName(resource.GetKind(), resource.GetName())
+	rb := &workv1alpha2.ResourceBinding{}
+	if err := d.Client.Get(context.TODO(), client.ObjectKey{Namespace: resource.GetNamespace(), Name: rbName}, rb); err != nil {
+		klog.Errorf("Failed to get ResourceBinding: %s, err: %+v", rbName, err)
+		return err
+	}
+
+	// 2. check whether rb related resource match to the given resource.
+	// the mismatch would never happen under normal circumstances, if it really happened, just print logs and return nil.
+	// this logic is also intended to be compatible with previous versions which talked about the issue of Ingress(networking.k8s.io/v1) and Ingress(extensions/v1beta1).
+	if rb.Spec.Resource.APIVersion != resource.GetAPIVersion() || rb.Spec.Resource.Kind != resource.GetKind() ||
+		rb.Spec.Resource.Name != resource.GetName() || rb.Spec.Resource.Namespace != resource.GetNamespace() {
+		klog.Errorf("The rb (%s) related resource is not match to the given resource: %s/%s", rbName, resource.GetNamespace(), resource.GetName())
+		return nil
+	}
+
+	// 3. clean the labels in binding
+	bindingCopy := rb.DeepCopy()
+	for _, label := range removeLabels {
+		delete(bindingCopy.Labels, label)
+	}
+	if err := d.Client.Update(context.TODO(), bindingCopy); err != nil {
+		klog.Errorf("Failed to remove binding labels, %s/%s, err: %+v", bindingCopy.GetNamespace(), bindingCopy.GetName(), err)
+		return err
+	}
+
+	// 4. clean the labels in resource
+	resourceCopy := resource.DeepCopy()
+	util.RemoveLabels(resourceCopy, removeLabels...)
+	if err := d.Client.Update(context.TODO(), resourceCopy); err != nil {
+		klog.Errorf("Failed to remove resource labels, %s/%s, err: %+v", resourceCopy.GetNamespace(), resourceCopy.GetName(), err)
+		return err
+	}
```

Comment:
```suggestion
	// 3. clean the labels in binding
	if err := d.CleanupResourceBindingLabels(rb, removeLabels...); err !=nil{
		return err
	}

	// 4. clean the labels in resource
	if err := d.CleanupLabels(rb.Spec.Resource, removeLabels...); err !=nil{
		return err
	}
```

Reply:
You are absolutely right, thank you very much for your advice~

Reply:
Oh, I will just pick `d. CleanupResourceBindingLabels`

`d.CleanupLabels` make things more complex.